File: chem_solution_mobile/lib/models/Achievement.dart
import 'package:chem_solution_mobile/assets/images.dart';
import 'package:chem_solution_mobile/models/Model.dart';
import 'MaterialGroup.dart';

class Achievement extends Model {
  int achievementId;
  String heading;
  String description;
  int moneyReward;
  int ratingReward;
  int countGoal;
  String imgAchievement;
  int materialGroupId;
  MaterialGroup materialGroup;
  // List<User> users = [];

  Achievement(
      {this.achievementId,
      this.heading,
      this.description,
      this.moneyReward,
      this.ratingReward,
      this.countGoal,
      this.materialGroupId,
      this.materialGroup,
      //  this.users,
      this.imgAchievement});

  @override
  Map<String, dynamic> toMap() {
    Map<String, dynamic> map = new Map<String, dynamic>();
    map['achievementId'] = achievementId;
    map['heading'] = heading;
    map['description'] = description;
    map['moneyReward'] = moneyReward;
    map['ratingReward'] = ratingReward;
    map['countGoal'] = countGoal;
    map['imgAchievement'] = imgAchievement;
    map['materialGroupId'] = materialGroupId;
    map['materialGroup'] = materialGroup.toMap();
    /*  List<Map<String, dynamic>> usersMaps = [];
    users.forEach((element) {
      usersMaps.add(element.toMap());
    });
    map['users'] = usersMaps; */
    return map;
  }

  static Achievement fromObject(dynamic o) {
    Achievement a = Achievement();
    a.achievementId = o['achievementId'];
    a.heading = o['heading'];
    a.imgAchievement = o['imgAchievement']?? notFound;
    a.description = o['description'];
    a.moneyReward = o['moneyReward'];
    a.ratingReward = o['ratingReward'];
    a.countGoal = o['countGoal'];
    a.materialGroupId = o['materialGroupId'];
   
    /*  
    
    a.materialGroup = MaterialGroup.fromObject(o['materialGroup']); */
    /*  o['users'].forEach((element) {
      a.users.add(User.fromObject(element));
    }); */
    return a;
  }
}

File: Models/Achievement.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Diagnostics.CodeAnalysis;

namespace ChemSolution.Models
{
    public class Achievement
    {
        [Key]
        public int AchievementId { get; set; }
        [StringLength(50, MinimumLength = 5)]
        public string Heading { get; set; }
        [StringLength(250, MinimumLength = 1)]
        public string ImgAchivement { set; get; }
        [StringLength(100, MinimumLength = 1)]
        public string Description { get; set; }
        [Range(0, int.MaxValue)]
        public int? MoneyReward { get; set; }
        [Range(0, int.MaxValue)]
        public int? RatingReward { get; set; }
        [Range(0, int.MaxValue)]
        public int? CountGoal { get; set; }
        public string ImgAchievemen { set; get; }
        public int? MaterialGroupId { get; set; }
        public MaterialGroup MaterialGroup { get; set; }
        public List<User> Users { set; get; } = new List<User>();
    }
}